packages/rum-core/src/browser/performanceCollection.ts

```diff
@@ -284,28 +284,31 @@ function computeRelativePerformanceTiming() {
   return result as RelativePerformanceTiming
 }
 
-function handlePerformanceEntries(lifeCycle: LifeCycle, configuration: RumConfiguration, entries: PerformanceEntry[]) {
-  entries.forEach((entry) => {
-    if (
-      entry.entryType === 'resource' ||
-      entry.entryType === 'navigation' ||
-      entry.entryType === 'paint' ||
-      entry.entryType === 'longtask' ||
-      entry.entryType === 'largest-contentful-paint' ||
-      entry.entryType === 'first-input' ||
-      entry.entryType === 'layout-shift'
-    ) {
-      handleRumPerformanceEntry(lifeCycle, configuration, (entry as unknown) as RumPerformanceEntry)
-    }
-  })
-}
+function handleRumPerformanceEntries(
+  lifeCycle: LifeCycle,
+  configuration: RumConfiguration,
+  entries: RumPerformanceEntry[]
+) {
+  const filteredEvents = entries.filter(
+    (entry) => !isIncompleteNavigation(entry) && !isForbiddenResource(configuration, entry)
+  )
 
-function handleRumPerformanceEntry(lifeCycle: LifeCycle, configuration: RumConfiguration, entry: RumPerformanceEntry) {
-  if (isIncompleteNavigation(entry) || isForbiddenResource(configuration, entry)) {
-    return
+  if (filteredEvents.length) {
+    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, filteredEvents)
   }
+}
 
-  lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED, entry)
+function filterRumPerformanceEntries(entries: PerformanceEntry[]) {
```

Comment:
`filterRumPerformanceEntries` filters first because it garantes that the rest of the code deals only with RumPerformanceEntry. Maybe a different naming would be better?

Reply:
> ‏I find it a bit confusing to have different levels of filtering.

+1

we could also have a `filterRumAllowedPerformanceEntries` that do all filtering operations

Reply:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Also you could use a `isRumPerformanceEntry(entry: PerformanceEntry): entry is RumPerformanceEntry` to have a nice typeguard without needing to cast `as unknown as RumPerformanceEntry`

Reply:
The type guard does not work because there is no relation between PerformanceEntry and RumPerformanceEntry. Also having a single filterRumAllowedPerformanceEntries where there is every check is weird  since we have to possible type of input PerformanceEntry[] and RumPerformanceEntry[].
The PerformanceEntry[] as to go through the first filter to be checked and cast to RumPerformanceEntry[].

Reply:
I changed the implementation according to what we said during the daily. Let me know what you think.